src/frontend/komponenter/Fagsak/Personlinje/Behandlingsmeny/HenleggBehandling/HenleggBehandling.tsx

```diff
@@ -0,0 +1,154 @@
+import useHenleggBehandling from './useHenleggBehandling';
+import { byggTomRessurs, RessursStatus } from '@navikt/familie-typer';
+
+interface IProps {
+    onListElementClick: () => void;
+    behandling: IBehandling;
+}
+
+interface HenleggelseÅrsakSelect extends HTMLSelectElement {
+    value: HenleggelseÅrsak | '';
+}
+
+const HenleggBehandling: React.FC<IProps> = ({ onListElementClick, behandling }) => {
+    const [visModal, settVisModal] = useState(false);
+
+    const {
+        fjernState,
+        onBekreft,
+        settSubmitRessurs,
+        submitRessurs,
+        selectedHenleggelseÅrsak,
+        settSelectedHenleggelseÅrsak,
+        begrunnelse,
+        settBegrunnelse,
+        valideringsFeil,
+        settValideringsfeil,
+    } = useHenleggBehandling(() => settVisModal(false));
+
+    const lukkHenleggBehandlingModal = () => {
+        fjernState();
+        settVisModal(false);
+    };
+
+    return (
+        <>
+            <KnappBase
+                mini={true}
+                onClick={() => {
+                    onListElementClick();
+                    settVisModal(true);
+                }}
+                disabled={
+                    ![
+                        BehandlingSteg.REGISTRERE_SØKNAD,
+                        BehandlingSteg.REGISTRERE_PERSONGRUNNLAG,
+                        BehandlingSteg.VILKÅRSVURDERING,
+                    ].includes(behandling.steg)
+                }
+            >
+                Henlegg behandling
+            </KnappBase>
+
+            <UIModalWrapper
+                modal={{
+                    actions: [
+                        <Flatknapp
+                            key={'avbryt'}
+                            mini={true}
+                            onClick={() => {
+                                lukkHenleggBehandlingModal();
+                            }}
+                            children={'Avbryt'}
+                        />,
+                        <Knapp
+                            key={'bekreft'}
+                            type={'hoved'}
+                            mini={true}
+                            onClick={() => onBekreft(behandling.behandlingId)}
+                            children={
+                                selectedHenleggelseÅrsak === HenleggelseÅrsak.SØKNAD_TRUKKET
+                                    ? 'Bekreft og send brev'
+                                    : 'Bekreft'
+                            }
+                            spinner={submitRessurs.status === RessursStatus.HENTER}
+                            disabled={submitRessurs.status === RessursStatus.HENTER}
+                        />,
+                    ],
+                    onClose: () => {
+                        lukkHenleggBehandlingModal();
+                    },
+                    lukkKnapp: true,
+                    tittel: 'Henlegg behandling',
+                    visModal,
+                }}
+            >
+                <SkjemaGruppe
```

Comment:
Her bør dere bruke den nye skjema hooken. Se BrevModulContext for eksempel. Brevskjema.tsx for bruk i komponent.